armi/reactor/tests/test_components.py

```diff
@@ -493,6 +493,15 @@ def test_changeNumberDensities(self):
         self.component.changeNDensByFactor(3.0)
         self.assertEqual(self.component.getNumberDensity("NA23"), 3.0)
 
+    def test_MassConserved(self):
```

Comment:
Can you explain in a bit more detail in the comments of this PR how this test would behave without these changes. Can you also extend this test to more components with different material types, like `Void` or a non-solid like `He` or `Sodium`? Just want to make sure that this test covers what it should and any edge cases

Reply:
Before, when input temperature density was used, the nDense * area were not equal. I am updating the the test to show how ARMI is expected to behave for a variety of cases.

This was the initial failing unit test i gave Tony to bug fix based on.

Reply:
Thanks for the addition! Unit tests are failing now because of a name issue but this test is much more detailed. I assume that you modify the ndens of Iron by dividing by the thermal expansion factor because that's emulating what happens on setting a new height if the component had a block it was attached to? That's my read on it and if so then I think this is good.

Reply:
yes, the comment was attempting to clarify how setBlock height modifies ndense of solid components

Reply:
Thanks!